base/testfiles/tlb-latexrelease-rollback-003-often.luatex.tlg

```diff
@@ -753,16 +819,52 @@ Skipping: [....-..-..] Standardise generic hook names on input line ....
 Skipping: [....-..-..] Standardise generic hook names on input line ....
 Skipping: [....-..-..] Standardise generic hook names on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
 Skipping: [....-..-..] Refuse setting rule for one-time hooks on input line ....
 Skipping: [....-..-..] Refuse setting rule for one-time hooks on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
 Skipping: [....-..-..] Standardise generic hook names on input line ....
 Skipping: [....-..-..] Standardise generic hook names on input line ....
 Skipping: [....-..-..] Standardise generic hook names on input line ....
 Skipping: [....-..-..] Standardise generic hook names on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
 Skipping: [....-..-..] Providing hooks on input line ....
 Skipping: [....-..-..] Providing hooks on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
+Skipping: [....-..-..] Hooks with args on input line ....
 Applying: [....-..-..] The hook management on input line ....
 END module: lthooks (....-..-..) on input line ....
+! LaTeX Error: Rollback code executed twice.
+See the LaTeX manual or LaTeX Companion for explanation.
+Type  H <return>  for immediate help.
+ ...                                              
+l. ...\cs:w __hook_rollback_tidying: \cs_end:
+If this was done on purpose in a testing environment, then there's (hopefully) to worry. Otherwise, something went wrong.
```

Comment:
nothing to worry?

Reply:
Yes.  This message was added because of this test file.  It loads `latexrelease` twice: first it rolls back to 0000/00/00, then forward again to 2019/10/01.  The code for `\@@_rollback_tidying:` is specifically for the case of rolling back to a date when there were no arguments in hooks (see commit 2f9ece5e), and that code can't be executed twice.  The error message essentially warns<sup>1</sup> you that `latexrelease` was loaded twice, and `\@@_rollback_tidying:` won't execute, so (hopefully) fine in a testing environment.

<sup>1</sup> Maybe it should be a warning instead of an error.